[ONNX] Clean up the diagnostics module

justinchuby opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #149864

Remove the diagnostics/SARIF module from ONNX exporter because it is obsolete unused.

cc @albanD

justinchuby committed:
Update

[ghstack-poisoned]

justinchuby committed:
Update

[ghstack-poisoned]

justinchuby committed:
Update

[ghstack-poisoned]

justinchuby committed:
Update

[ghstack-poisoned]

malfet commented:
@pytorchbot revert -m "This indeed broke Mac testing see https://hud.pytorch.org/hud/pytorch/pytorch/1c98dc3664bc187d7529316fdc141fcdbf13d024/1?per_page=50&name_filter=macos&mergeLF=true" -c ignoredsignal

malfet commented:
@justinchuby just curious, why do you think failures were unrelated?

justinchuby commented:
Hmm thanks for noticing that. It's strange that only macOS is failing and it does seem to be caused by this change at the first sight. Will look into it.

justinchuby committed:
Update on "[ONNX] Clean up the diagnostics module"


Remove the diagnostics/SARIF module from ONNX exporter because it is obsolete unused.

cc albanD

[ghstack-poisoned]

justinchuby commented:
@pytorchbot merge -i

justinchuby commented:
@malfet I fixed the issue by hiding references to onnx.TensorProto. Not sure know how the error happened in the first place because this change didn't touch that part. But it should be fixed nevertheless.
